etl/steps/data/garden/un/2024-07-16/migrant_stock.py

```diff
@@ -0,0 +1,362 @@
+    tb_d_share = format_table(tb_d_share, ["country"], "migrant_share")
+
+    ## data on origin
+    tb_o = format_table(tb_o, ["country"], "migrants")
+
+    ## data on sex and age
+    sa_share_cols_rename = {key: "share_of_" + value for key, value in SA_COLS_RENAME.items()}
+
+    # rename columns
+    tb_sa_total = tb_sa_total.rename(columns=SA_COLS_RENAME)
+    tb_sa_share = tb_sa_share.rename(columns=sa_share_cols_rename)
+
+    # change dtype to numeric
+    for col in SA_COLS_RENAME.values():
+        tb_sa_total[col] = pd.to_numeric(tb_sa_total[col], errors="coerce")
+
+    tb_sa_total = add_metadata(tb_sa_total, list(SA_COLS_RENAME.values()), "year")
+
+    # drop total columns (they add up to 100)
+    tb_sa_total = tb_sa_total.drop(columns=["total", "total_1", "total_2"])
+    tb_sa_share = tb_sa_share.drop(columns=["total", "total_1", "total_2"])
+
+    # harmonize country names
+    tb_sa_total = geo.harmonize_countries(
+        df=tb_sa_total, countries_file=paths.country_mapping_path, country_col="country", warn_on_unused_countries=False
+    )
+    tb_sa_share = geo.harmonize_countries(
+        df=tb_sa_share, countries_file=paths.country_mapping_path, country_col="country", warn_on_unused_countries=False
+    )
+
+    # remove duplicate data
+    tb_sa_total = tb_sa_total.drop_duplicates()
+    tb_sa_share = tb_sa_share.drop_duplicates()
+
+    ## population data
+    for col in tb_pop.columns[2:]:
+        tb_pop[col] = pd.to_numeric(tb_pop[col], errors="coerce")
+        tb_pop[col] = tb_pop[col] * 1000
+    tb_pop = geo.harmonize_countries(
+        df=tb_pop, countries_file=paths.country_mapping_path, country_col="country", warn_on_unused_countries=False
+    )
+    tb_pop = tb_pop.drop_duplicates()
+
+    ## combine tables except for destination and origin table
+    # rename columns to differentiate between values
+    tb_d_total = tb_d_total.rename(
+        columns={
+            "migrants_all_sexes": "immigrants_all",
+            "migrants_female": "immigrants_female",
+            "migrants_male": "immigrants_male",
+        }
+    )
+
+    tb_d_share = tb_d_share.rename(
+        columns={
+            "migrant_share_all_sexes": "immigrant_share_of_dest_population_all",
+            "migrant_share_female": "immigrant_share_of_dest_population_female",
+            "migrant_share_male": "immigrant_share_of_dest_population_male",
+        }
+    )
+
+    tb_o = tb_o.rename(
+        columns={
+            "migrants_all_sexes": "emigrants_all",
+            "migrants_female": "emigrants_female",
+            "migrants_male": "emigrants_male",
+        }
+    )
+
+    tb = pr.multi_merge(
+        [tb_d_total, tb_d_share, tb_o, tb_sa_total, tb_sa_share, tb_pop], on=["country", "year"], how="outer"
+    )
+    tb.metadata.short_name = "migrant_stock"
+
+    ## Calculate missing values:
+    # under 15 y/o migrants, under 20 y/o migrants
+    tb["immigrants_under_15"] = (
```

Comment:
you mean in a comment, or somehow programmatically? In this case, yes, if any of these values are nan, I would like <15 to be nan as well, since it is otherwise not comparable with the values for other countries. I think in practice since these are estimates, all of these values are filled in the original table.

Reply:
Ideally both. You can mention in the comment if you want the sum to be nan if any of the added columns has a nan. If that's the case, I think summing them this way will do the job (which may not necessarily be the case if you did `tb[columns].sum(axis=1)`, if I remember correctly).
Sometimes one of the columns you are adding up is not that important and has more sparse data, so you may want to ignore their nans (or fill them with something).

Reply:
Thank you! That is good to know - I added a comment for clarity!